test/src/utils/kalman_multivariate.test.cpp

```diff
@@ -0,0 +1,227 @@
+  VectorXf x1 = VectorXf::Random(n);
+  VectorXf z = VectorXf::Random(n);
+  MatrixXf P0 = MatrixXf::Zero(n, n);
+  MatrixXf A = MatrixXf::Random(n, n);
+  MatrixXf B = MatrixXf::Random(n, k);
+  MatrixXf Q = MatrixXf::Random(n, n);
+  MatrixXf H = MatrixXf::Random(m, n);
+  MatrixXf R = MatrixXf::Random(m, m);
+  MatrixXf P = MatrixXf::Random(n, n);
+  std::string zero_state_estimate_err = "Should handle zero vector as state estimate";
+  std::string arb_state_estimate_err = "Should handle any arbitrary vector as state estimate";
+  std::string zero_covariance_err = "Should handle zero state covariance";
+  std::string arb_covariance_err = "Should handle any arbitrary state covariance";
+
+  void SetUp()
+  {
+    // Assigns the matrices and vectors to the properties within the kalman object.
+    kalman.setModels(A, Q, H, R);
+    kalman.setInitial(x0, P);
+  }
+  void TearDown() {}
+};
+
+/**
+ * Test fixture used for determining whether Kalman handles the zero vector as input for the state
+ * estimates. Should accept
+ */
+TEST_F(KalmanFunctionality, handlesZeroStateEstimate)
+{
+  ASSERT_EQ(kalman.getStateEstimate(), x0) << zero_state_estimate_err;
+}
+
+/**
+ * Test fixture used for determining whether Kalman handles an arbitrary vector (using Random()) as
+ * input for the state estimates. This should always be accepted.
+ */
+TEST_F(KalmanFunctionality, handlesArbitraryStateEstimate)
+{
+  kalman.setInitial(x1, P);
+  ASSERT_EQ(kalman.getStateEstimate(), x1) << arb_state_estimate_err;
+}
+
+/**
+ * Test fixture used for determining whether Kalman handles the zero matrix as input for the state
+ * covariance. This should always be accepted.
+ */
+TEST_F(KalmanFunctionality, handlesZeroStateCovariance)
+{
+  kalman.setInitial(x0, P0);
+  ASSERT_EQ(kalman.getStateCovariance(), P0) << zero_covariance_err;
+}
+
+/**
+ * Test fixture used for determining whether Kalman handles an arbitrary matrix (using Random()) as
+ * input for the state covariance. This should always be accepted.
+ */
+TEST_F(KalmanFunctionality, handlesArbitraryStateCovariance)
+{
+  ASSERT_EQ(kalman.getStateCovariance(), P) << arb_covariance_err;
+}
+
+// -------------------------------------------------------------------------------------------------
+// Mathematical Properties
+// -------------------------------------------------------------------------------------------------
+
+/**
+ * Struct used for testing the mathematical properties of kalman_multivariate. Constructor initiates
+ * an array of Vectors and Matrices to store arbitrary vectors for testing purposes. The arrays are
+ * then populated with values so they can be tested.
+ */
+struct KalmanMathematics : public::testing::Test {
+  unsigned int n = 3;
+  unsigned int m = 2;
+  unsigned int k = 1;
+
+ protected:
+  KalmanMultivariate kalmanMathWithoutControl = KalmanMultivariate(n, m, 0);
+  KalmanMultivariate kalmanMathWithControl = KalmanMultivariate(n, m, k);
+  VectorXf x1_Data[100];
```

Comment:
I'm not sure about the unnamed constant `100`. I know very little about Kalman filters and it's not obvious to me whether this is an arbitrary testing value or essential for this code to work.